at_end2end_test/test/cached_key_test.dart

```diff
@@ -132,6 +132,70 @@ void main() {
     expect(response, contains('data:$value'));
   }, timeout: Timeout(Duration(seconds: 100)));
 
+  /// The purpose of this test verify the following:
+  /// 1. Share a key from atsign_1 to atsign_2 with ttr, with autoNotify:true
+  /// 2. lookup from atsign_2 returns the correct value
+  /// 3.  Set the autoNotify to false using the config verb 
+  /// 4. Update the existing key to a new value 
+  /// 4. lookup with bypass_cache set to true should return the updated value
+  /// 5. lookup with bypass_cache set to false should return the old value
+  test('update-lookup verb passing bypasscache ', () async {
+    ///Update verb on atsign_1
+    var oldValue = 'Hyderabad';
+
+    await sh1
+        .writeCommand('update:ttr:100000:$atSign_2:fav-city$atSign_1  $oldValue');
+    String response = await sh1.read();
+    print('update verb response : $response');
+    assert(
+        (!response.contains('Invalid syntax')) && (!response.contains('null')));
+
+    ///lookup verb alice  atsign_2
+    await sh2.writeCommand('lookup:fav-city$atSign_1');
+    response = await sh2.read();
+    print('lookup verb response : $response');
+    expect(response, contains('data: $oldValue'));
+
+    // config set auto notify to false
+    await sh1.writeCommand('config:set:autoNotify=false');
+    response = await sh1.read();
+    print('config set verb response is $response');
+    expect(response, contains('data:ok'));
+
+    var newValue = 'Chennai';
+    await sh1.writeCommand('update:$atSign_2:fav-city$atSign_1  $newValue');
+    response = await sh1.read();
+    print('update verb response : $response');
+    assert(
+        (!response.contains('Invalid syntax')) && (!response.contains('null')));
+
+    ///lookup should return the old value
+    await sh2.writeCommand('lookup:fav-city$atSign_1');
+    response = await sh2.read();
+    print('lookup verb response : $response');
+    expect(response, contains('data: $oldValue'));
+
+    /// lookup with bypass_cache set to true
+    /// should return the newly updated value
+    await sh2.writeCommand('lookup:bypassCache:true:fav-city$atSign_1');
+    response = await sh2.read();
+    print('lookup verb response : $response');
+    expect(response, contains('data: $newValue'));
+
+    /// lookup with bypass_cache set to false
+    /// should return the old value
+    await sh2.writeCommand('lookup:bypassCache:false:fav-city$atSign_1');
+    response = await sh2.read();
+    print('lookup verb response : $response');
+    expect(response, contains('data: $oldValue'));
+
+    // reset the autoNotify to default
+    await sh1.writeCommand('config:reset:autoNotify');
```

Comment:
This should be in a finally {} block so that even if the test assertions fail, we still set autoNotify back to what it should be

Reply:
Added the code snippet to the finally block.